specreduce/extract.py

```diff
@@ -43,22 +43,37 @@ def _get_boxcar_weights(center, hwidth, npix):
         A 2D image with weights assigned to pixels that fall within the
         defined aperture.
     """
-    weights = np.zeros(npix)
-
-    # shift center from integer to pixel space, where pixel N is [N-0.5, N+0.5),
-    # not [N, N+1). a pixel's integer index corresponds to its middle, not edge
-    center += 0.5
-
-    # pixels given full weight because they sit entirely within the aperture
-    fullpixels = [max(0, int(np.ceil(center - hwidth))),
-                  min(int(np.floor(center + hwidth)), npix)]
-    weights[fullpixels[0]:fullpixels[1]] = 1
-
-    # pixels at the edges of the boxcar with partial weight, if any
-    if fullpixels[0] > 0:
-        weights[fullpixels[0] - 1] = hwidth - (center - fullpixels[0])
-    if fullpixels[1] < npix:
-        weights[fullpixels[1]] = hwidth - (fullpixels[1] - center)
+    weights = np.zeros((npix))
+    if hwidth == 0:
+        # the logic below would return all zeros anyways, so might as well save the time
+        # (negative widths should be avoided by earlier logic!)
+        return weights
```

Comment:
this shouldn't ever actually be called (so isn't being covered by tests), but in case we use the internal function somewhere without a width check, I don't think the redundancy hurts